### PR TITLE
Add toJSON method to EditSession

### PR DESCRIPTION
**Pull request by vanillajonathan**
*Issue #, if available:* #1452

*Description of changes:*
Adds a [`toJSON`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior) method to facilitate serialization of the the `EditSession` object into JSON through the [`JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) function.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/edit_session.js
Comment (andredcoliveira) [inline src/edit_session.js:279]: I think we can combine this with #5236 and make this:

```suggestion
            history: this.getUndoManager().toJSON()
```

File: src/edit_session.js
Comment (andredcoliveira) [inline src/edit_session.js:286]: Could you remove the semicolon to make the build happy? :)

```suggestion
    }
```

**Review by andredcoliveira — COMMENTED** [review]
Had a look; could you add some basic testing to ensure the output is as expected?

File: src/edit_session.js
Comment (vanillajonathan) [reply src/edit_session.js:279]: Yeah that sounds good, but then that PR would need to be merged first I guess.

Comment (andredcoliveira) [reply src/edit_session.js:308]: That's correct. Would you be able to merging these with @NotLazy? They mentioned they're going to add tests when they get a chance and then we should be able to merge it.

Comment (vanillajonathan) [reply src/edit_session.js:308]: Yeah, we can wait until he adds the tests then you can merge his PR, and I can adjust my PR and we can merge this. Either that, or you can merge this now, then when has added tests and his PR is merged, then you, he or I can file a new PR that changes the `history` property to use his code.

Comment (vanillajonathan) [reply src/edit_session.js:308]: @andredcoliveira can you merge this?

**Review by InspiredGuy — COMMENTED** [review]
looks good, just couple of comments about return type

File: src/edit_session.js
Comment (vanillajonathan) [reply src/edit_session.js]: Done

File: ace.d.ts
Comment (vanillajonathan) [reply ace.d.ts]: Done

**Review by InspiredGuy — APPROVED** [review]

**Comment by InspiredGuy** [comment]
Thank you for finalising this

**Pull request merged**
